### ISSUE TITLE
Barrel Smoke

### ISSUE THREAD

**Issue by TeslaCoil9001** [studio-minus/ppg-bugs-and-requests#2067]
**Feature description**
When a gun is fired for a sustained time, the barrel will start to smoke slightly. This also happens when the LMG's barrel gets hot and will happen every time the flintlock pistol is added.

**Feature purpose**
Polish and realism. It could add a lot of visual flair to the game. It may be performance heavy, so it should only be enabled if Fancy Effects is turned on.

**Additional context**
![image](https://user-images.githubusercontent.com/112362312/188257369-e91dad4a-1eb5-4831-bf7e-6082b6a5f130.png)

**Comment by ATPGaming** [comment]
good idea and it can also be on revolvers when they fire, but only a small amount of smoke

**Comment by TeslaCoil9001** [comment]
> good idea and it can also be on revolvers when they fire, but only a small amount of smoke

Well revolvers only smoke if you shoot them rapidly, if your revolver smokes after being fired once you might want to get one that was actually created in the last 70 years

**Comment by ATPGaming** [comment]
I just realized, is a duplicate of #474
https://github.com/studio-minus/ppg-bugs-and-requests/issues/474

**Comment by DindinYT37** [comment]
> > I just realized, is a duplicate of #474 #474
> 
> Well that is technically suggesting it for every time guns are fired not for a sustained time

It's subjective. Though this would mean it is related, which is precisely why I always say "Duplicate of or Related to" and not just one of the two.

**Comment by ATPGaming** [comment]
I just observed that when laser pointers get hot they also smoke, so Zooi could reuse the particles for barrel smoke

**Comment by ATPGaming** [comment]
There's this particle effect that Zooi could reuse for the barrel smoke and it would look very good
https://user-images.githubusercontent.com/109692890/200432310-f1e922db-29bc-4751-9672-ba22aac2253f.mp4
